wooting-macro-backend/src/plugin/mouse.rs

```diff
@@ -80,43 +43,36 @@ pub enum MousePressAction {
 
 impl MouseAction {
     /// Creates a new MouseAction from a rdev event and sends it to the channel for async execution.
-    pub async fn execute(&self, send_channel: UnboundedSender<EventType>) {
+    pub async fn execute(&self, send_channel: UnboundedSender<rdev::EventType>) -> Result<()> {
         match &self {
             MouseAction::Press { data } => match data {
                 MousePressAction::Down { button } => {
-                    send_channel
-                        .send(rdev::EventType::ButtonPress(button.into()))
-                        .unwrap();
+                    send_channel.send(rdev::EventType::ButtonPress(button.into()))?;
                 }
                 MousePressAction::Up { button } => {
-                    send_channel
-                        .send(rdev::EventType::ButtonRelease(button.into()))
-                        .unwrap();
+                    send_channel.send(rdev::EventType::ButtonRelease(button.into()))?;
                 }
                 MousePressAction::DownUp { button, duration } => {
-                    send_channel
-                        .send(rdev::EventType::ButtonPress(button.into()))
-                        .unwrap();
+                    send_channel.send(rdev::EventType::ButtonPress(button.into()))?;
 
                     tokio::time::sleep(time::Duration::from_millis(*duration as u64)).await;
 
-                    send_channel
-                        .send(rdev::EventType::ButtonRelease(button.into()))
-                        .unwrap();
+                    send_channel.send(rdev::EventType::ButtonRelease(button.into()))?;
                 }
             },
 
             MouseAction::Move { x, y } => {
-                let display_size = rdev::display_size().unwrap();
+                let display_size = rdev::display_size().map_err(|err| {
+                    anyhow::Error::msg(format!("Error getting displays: {:?}", err))
+                })?;
```

Comment:
Shouldn't you be able to do this cleaner by using `.context` from Anyhow?

Reply:
that may apply for a lot of these then, I can check for all of them and see where it makes sense to try using the .context. I only tweaked src-tauri/main before

Reply:
Tried, but the library and subsequently other libraries where I use map_err don't have proper implementation for anyhow error and or display, trait bounds unsatisfied. This was the only way to convert it to a string and format it

Reply:
Fun